csrc/multidevice/communication.h

```diff
@@ -68,16 +69,35 @@ class Communication {
       Communicator& comm,
       std::optional<CommunicatorBackend> backend = std::nullopt) = 0;
 
+  // Allocates a contiguous intermediate tensor.
+  // Required for NCCL/UCC which expect contiguous input/output buffers.
+  // input tensor was sharded like [a b .. DIDx ... c] where DIDx is the axis
+  // parallelized on DIDx the output tensor elements will be ordered as
+  //  [DIDx a b c]  with device dimensions pushed to the outer most axis
+  virtual bool requiresRelayoutOutputTensor(
```

Comment:
Why not making these method private ? In my mind they are internal method and shouldn't be part of the Communication API

Reply:
Also, why making them virtual ?